pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -1455,6 +1452,28 @@ default 'raise'
             np.array([self.value], dtype="i8"), tz=own_tz)
         return Timestamp(normalized[0]).tz_localize(own_tz)
 
+    def strftime(self, format: str) -> str:
+        """
+        Constructs datetime style `format` string from Timestamp.
+
+        See `datetime <https://docs.python.org/3/library/datetime\
+.html#strftime-and-strptime-format-codes>`_ module for all available directives.
+
+        Parameters
+        ----------
+        format : str
+            String of formatting directives
+
+        Returns
+        -------
+        str
+            String representation of Timestamp
+        """
+        if '%f' in format and self.nanosecond:
+            replacement = f'{self.microsecond * 1000 + self.nanosecond:09d}'
+            format = re.sub('%f', replacement, format)
```

Comment:
can you use string replacement rather than re? doesn't this change perf?

Reply:
Good call. Tested both and string replace is WAY more performant.